runelite-client/src/main/java/net/runelite/client/plugins/wintertodt/WintertodtPlugin.java

```diff
@@ -0,0 +1,478 @@
+import net.runelite.api.events.GameTick;
+import net.runelite.api.events.ItemContainerChanged;
+import net.runelite.api.events.SetMessage;
+import net.runelite.client.Notifier;
+import net.runelite.client.chat.ChatMessageManager;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import net.runelite.client.ui.overlay.OverlayManager;
+
+@PluginDescriptor(
+	name = "Wintertodt",
+	description = "Wintertodt"
+)
+@Slf4j
+public class WintertodtPlugin extends Plugin
+{
+	static final int WINTERTODT_REGION = 6462;
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private Client client;
+
+	@Inject
+	private OverlayManager overlayManager;
+
+	@Inject
+	private WintertodtOverlay overlay;
+
+	@Inject
+	private WintertodtConfig config;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Getter(AccessLevel.PACKAGE)
+	private WintertodtActivity currentActivity = WintertodtActivity.IDLE;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int inventoryScore;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int totalPotentialinventoryScore;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int numLogs;
+
+	@Getter(AccessLevel.PACKAGE)
+	private int numKindling;
+
+	private Instant lastActionTime;
+
+	@Getter(AccessLevel.PACKAGE)
+	private boolean isInWintertodt;
+
+	@Provides
+	WintertodtConfig getConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(WintertodtConfig.class);
+	}
+
+	@Override
+	protected void startUp() throws Exception
+	{
+		reset();
+		overlayManager.add(overlay);
+	}
+
+	@Override
+	protected void shutDown() throws Exception
+	{
+		overlayManager.remove(overlay);
+		reset();
+	}
+
+	private void reset()
+	{
+		inventoryScore = 0;
+		totalPotentialinventoryScore = 0;
+		numLogs = 0;
+		numKindling = 0;
+		currentActivity = WintertodtActivity.IDLE;
+	}
+
+	private boolean isInWintertodtRegion()
+	{
+		if (client.getLocalPlayer() != null)
+		{
+			return client.getLocalPlayer().getWorldLocation().getRegionID() == WINTERTODT_REGION;
+		}
+
+		return false;
+	}
+
+	@Subscribe
+	public void onGameTick(GameTick gameTick)
+	{
+		if (!isInWintertodtRegion())
+		{
+			if (isInWintertodt)
+			{
+				log.debug("Left Wintertodt!");
+				reset();
+			}
+
+			isInWintertodt = false;
+			return;
+		}
+
+		if (!isInWintertodt)
+		{
+			reset();
+			log.debug("Entered Wintertodt!");
+		}
+		isInWintertodt = true;
+
+		checkActionTimeout();
+	}
+
+	private void checkActionTimeout()
+	{
+		if (currentActivity == WintertodtActivity.IDLE)
+			return;
+
+		int currentAnimation = client.getLocalPlayer() != null ? client.getLocalPlayer().getAnimation() : -1;
+		if (currentAnimation != IDLE)
+		{
+			return;
+		}
+
+		Duration actionTimeout = Duration.ofSeconds(3);
+		Duration sinceAction = Duration.between(lastActionTime, Instant.now());
+
+		if (sinceAction.compareTo(actionTimeout) >= 0)
+		{
+			log.debug("Activity timeout!");
```

Comment:
Can this debug log be at least more descriptive?

Reply:
The fact its from the wintertodt logger is probably enough?

Reply:
Maybe, but I would probably also log what activity timeouted